demos/ota/ota_demo_core_http/ota_demo_core_http.c

```diff
@@ -0,0 +1,1620 @@
+/*
+ * Copyright (C) 2020 Amazon.com, Inc. or its affiliates.  All Rights Reserved.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of
+ * this software and associated documentation files (the "Software"), to deal in
+ * the Software without restriction, including without limitation the rights to
+ * use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of
+ * the Software, and to permit persons to whom the Software is furnished to do so,
+ * subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+ * FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+ * COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+ * IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+/**
+ * @file ota_demo_core_mqtt.c
+ * @brief OTA update example using coreMQTT.
+ */
+
+/* Standard includes. */
+#include <assert.h>
+#include <stdlib.h>
+#include <stdbool.h>
+#include <errno.h>
+
+/* Include Demo Config as the first non-system header. */
+#include "demo_config.h"
+
+/* OpenSSL sockets transport implementation. */
+#include "openssl_posix.h"
+
+/* Clock for timer. */
+#include "clock.h"
+
+/* pthread include. */
+#include <pthread.h>
+#include <semaphore.h>
+
+/* MQTT include. */
+#include "core_mqtt.h"
+#include "mqtt_subscription_manager.h"
+
+/* HTTP include. */
+#include "core_http_client.h"
+
+/* Common HTTP demo utilities. */
+#include "http_demo_utils.h"
+
+/*Include backoff algorithm header for retry logic.*/
+#include "backoff_algorithm.h"
+
+/* OTA Library include. */
+#include "ota.h"
+#include "ota_config.h"
+
+/* OTA Library Interface include. */
+#include "ota_os_posix.h"
+#include "ota_mqtt_interface.h"
+#include "ota_pal_posix.h"
+
+/* Include firmware version struct definition. */
+#include "ota_appversion32.h"
+
+/**
+ * These configuration settings are required to run the OTA demo which uses mutual authentication.
+ * Throw compilation error if the below configs are not defined.
+ */
+#ifndef AWS_IOT_ENDPOINT
+    #error "Please define AWS IoT MQTT broker endpoint(AWS_IOT_ENDPOINT) in demo_config.h."
+#endif
+#ifndef ROOT_CA_CERT_PATH
```

Comment:
The check for the HTTP root CA cert also needs to be added

Reply:
Addressed in https://github.com/aws/aws-iot-device-sdk-embedded-C/pull/1425/commits/9f18e9905580d6e4067eaa4a215da9dbe1e2453a